xds/src/test/java/io/grpc/xds/WeightedRoundRobinLoadBalancerTest.java

```diff
@@ -876,9 +1107,24 @@ private static class FakeSocketAddress extends SocketAddress {
 
   private static class FakeRandom extends Random {
     @Override
-    public double nextDouble() {
+    public int nextInt() {
       // return constant value to disable init deadline randomization in the scheduler
-      return 0.322023;
+      // sequence will be initialized to 0
```

Comment:
In this case, you can reduce the boilerplate code by adding a constructor in FakeRandom and remove the other two  classes.